### ISSUE TITLE
Javadoc generation

### ISSUE THREAD

**Issue by Gustorn** [gregory094/softlab4#45]
As of #40 we now support JavaDoc to LaTeX generation. This has two (potential) issues:

1. It inserts the JavaDoc generated content at the beginning of the documentation. I feel like adding it to the end would be a better default.
2. It's in English, while the rest of the document is in Hungarian. After a bit of research I believe JavaDoc translations are commonly done with a custom doclet, but unfortunately we already use one for the LaTeX generation. Can we modify it somehow, or is this not an issue?

**Comment by nyari** [comment]
1. I've already removed it from the beginning. If my addition to the
documentation will pass. We'll only need later on in the design process.
Somewhere around the end of the "Skeleton" milestone. Until then it will
lay dormant in the repository until we explicitly need it in the
documentation.
2.1. If there is a doclet that does the translation then it is possible to
merge the two doclets somehow provided we have the sources. (TexDoclet
source is available) This is definitely something worth considering but may
be more trouble than it's worth. But we'll have to ask our consultant about
this as well. I think if he gives it a green flag then we should not
bother.
2.2. Different solutions would be:
2.2.1 Finding a JavaDoc to LaTeX doclet that has a translation option
2.2.2 Writing a script that translates the LaTeX output of TexDoclet

**Comment by Gustorn** [comment]
Relevant [SO question](http://stackoverflow.com/questions/12074244/how-to-generate-a-translated-javadoc-with-javadoc-exe-not-the-content-but-the). I tried the steps and they are fairly easy. The unfortunate consequence is that it makes doc generation non-portable (it won't generate Hunargian output on the lab PCs). If we want to make a fully Hungarian doc, it probably wouldn't be that difficult, the 2 files in question are ~800 lines.

**Comment by Gustorn** [comment]
According to [TexDoclet](http://doclet.github.io/) you can actually use Markdown syntax in your JavaDoc comments. Any feelings about preferring that to XML? It's much more concise and it's actually human readble even when not processed. As a comparison:

Ordered list in regular JavaDoc
````
/**
 * <ol>
 * <li> First list item </li>
 * <li> Second list item </li>
 * </ul>
 */
````

And in Markdown:
````
/**
 * 1. First list item
 * 2. Second list item
 */
````

This should produce the same output.

**Comment by Gustorn** [comment]
I fixed some of the JavaDoc generation in #141, but there's still no docs generated for the subpackages in `commands`.
